devices/HapticGlove/src/HapticGlove.cpp

```diff
@@ -0,0 +1,621 @@
+                                                                                                                pImpl->hapticActuatorPrefix + pImpl->gloveData.humanFingerNames[i] + "::ForceFeedback" )});
+    }
+
+    for(size_t i=0; i < pImpl->gloveData.humanFingerNames.size();i++)
+    {
+        pImpl->sensegloveHapticActuatorVector.push_back(ElementPtr<SenseGloveImpl::SenseGloveHapticActuator>{std::make_shared<SenseGloveImpl::SenseGloveHapticActuator>(pImpl.get(),
+                                                                                                                pImpl->hapticActuatorPrefix + pImpl->gloveData.humanFingerNames[i] + "::VibroTactileFeedback" )});
+    }
+
+    for(size_t i=0; i<pImpl->gloveData.humanJointNames.size(); i++)
+        pImpl->gloveData.humanJointSensorNameIdMap.emplace(std::make_pair(pImpl->jointSensorPrefix + pImpl->gloveData.humanJointNames[i], i));
+
+    for(size_t i=0; i<pImpl->gloveData.humanFingerNames.size(); i++)
+        pImpl->gloveData.humanHapticActuatorNameIdMap.emplace(std::make_pair(pImpl->hapticActuatorPrefix + pImpl->gloveData.humanFingerNames[i] + "::ForceFeedback", i));
+
+    for(size_t i=0; i<pImpl->gloveData.humanFingerNames.size(); i++)
+        pImpl->gloveData.humanHapticActuatorNameIdMap.emplace(std::make_pair(pImpl->hapticActuatorPrefix + pImpl->gloveData.humanFingerNames[i] + "::VibroTactileFeedback", i));
+
+    yInfo()<<LogPrefix<<"The device is opened successfully.";
+    
+    // Start the PeriodicThread loop
+    if (!start()) {
+        yError() << LogPrefix << "Failed to start the period thread.";
+        return false;
+    }
+
+
+    return true;
+}
+
+// ===========================================
+// SenseGlove implementation of VirtualLinkKinSensor
+// ===========================================
+
+class HapticGlove::SenseGloveImpl::SenseGloveVirtualLinkKinSensor : public sensor::IVirtualLinkKinSensor
+{
+public:
+    SenseGloveVirtualLinkKinSensor( HapticGlove::SenseGloveImpl* gloveImplPtr,
+                                    const sensor::SensorName& name = {},
+                                    const sensor::SensorStatus& status = sensor::SensorStatus::Ok)
+                                  : IVirtualLinkKinSensor(name, status),
+                                    m_gloveImpl(gloveImplPtr)
+    { }
+
+    ~SenseGloveVirtualLinkKinSensor() override = default;
+
+    bool getLinkAcceleration(Vector3& linear, Vector3& angular) const override
+    {
+        // we do not handle linear and angular accelerations in the current implementation
+        linear.fill(0.0);
+        angular.fill(0.0);
+
+        return true;
+    }
+
+    bool getLinkPose(Vector3& position, Quaternion& orientation) const override
+    {
+        // we do not handle position in the current implementation
+        position.fill(0.0);
+
+        assert(m_gloveImpl != nullptr);
+
+        std::lock_guard<std::mutex> lock(m_gloveImpl->mutex);
```

Comment:
Should this mutex also protect the position? Even if it is not currently handled. Also, I'd use the mutex is all `get*` methods of all sensor implementations.

Reply:
done.